### PR TITLE
Setting up a placeholder api with deployment

### PR DESCRIPTION
**Pull request by kstigen**

### CODE REVIEW

File: keycloak/keycloak.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	clients, err := k.AdminClient.GetClients(k.Ctx, token.AccessToken, k.KeycloakOpenEpiRealm, gocloak.GetClientsParams{})
+	if err != nil {
+		return nil, models.ClientLookupErr
+	}
+
+	// Filter list of clients by UserName
+	clientList := make([]*models.Client, 0)
+	for _, cl := range clients {
+		if s.HasPrefix(*cl.ClientID, UserName) {
+			clientList = append(clientList, &models.Client{InternalID: cl.ID, ClientID: cl.ClientID, ClientName: cl.Name, ClientSecret: cl.Secret})
```
Comment (daniejoh) [inline keycloak/keycloak.go]: Could you have used lo.Filter here?
```
even := lo.Filter(clients, func(client models.Client, index int) bool {
    return s.HasPrefix(*client.ClientID, UserName)
})
```
I think it makes it more readable.

File: deployment/argocd/app.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - resources-finalizer.argocd.argoproj.io
+spec:
+  destination:
+    namespace: apps
+    server: https://kubernetes.default.svc
+  project: apps
+  source:
+    path: deployment/kubernetes/
+    repoURL: https://github.com/openearthplatforminitiative/client-registration-api
+    targetRevision: Issue-253-Simplest-possible-API
```
Comment (daniejoh) [inline deployment/argocd/app.yaml]: Change this to HEAD.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,30 @@
...
+```
+with-expecter: True
+dir: tests/mocks/{{.InterfaceDirRelative}}
+mockname: "Mock{{.InterfaceName}}"
+outpkg: "{{.PackageName}}"
+filename: "mock_{{.InterfaceName}}.go"
+all: True
+packages:
+  github.com/openearthplatforminitiative/client-registration-api/keycloak:
+```
```
Comment (daniejoh) [inline README.md]: Nitpicky, but you have pushed .mockery.yaml, can you refer to that file instead?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,30 @@
 # client-registration-api
 API for registering application clients for users of OpenEPI
+
+## Generating mocks for unit testing
+We use the tool mockery for generating mocks for unit testing. To install mockery, run the following command:
+```
+brew install mockery
```
Comment (daniejoh) [inline README.md]: Not everyone might use brew. Refer to installation docs instead? https://vektra.github.io/mockery/latest/installation/#installation

File: deployment/kubernetes/client-registration-api.yaml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    matchLabels:
+      app: client-registration-api
+  template:
+    metadata:
+      labels:
+        app: client-registration-api
+#      annotations:
+#        prometheus.io/scrape: "true"
+#        prometheus.io/port: "8080"
+#        prometheus.io/path: "/metrics"
```
Comment (daniejoh) [inline deployment/kubernetes/client-registration-api.yaml:18]: Are we dropping logs and metrics for now?

File: keycloak/keycloak.go [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+	clients, err := k.AdminClient.GetClients(k.Ctx, token.AccessToken, k.KeycloakOpenEpiRealm, gocloak.GetClientsParams{})
+	if err != nil {
+		return nil, models.ClientLookupErr
+	}
+
+	// Filter list of clients by UserName
+	clientList := make([]*models.Client, 0)
+	for _, cl := range clients {
+		if s.HasPrefix(*cl.ClientID, UserName) {
```
Comment (daniejoh) [inline keycloak/keycloak.go]: I think you should avoid import aliasing here, and just use `strings.HasPrefix` instead of `s.HasPrefix`.

Comment (daniejoh) [reply keycloak/keycloak.go]: Although we are not following this style guide, I agree with the statement that "In all other scenarios, import aliases should be avoided unless there is a direct conflict between imports." https://github.com/uber-go/guide/blob/master/style.md#import-aliasing 

What do you think?

File: tests/unit/data/test_data.go [hunk 7]
```diff
@@ -0,0 +1,55 @@
+package data
+
+import (
+	"encoding/json"
+	"fmt"
+	ut "github.com/go-playground/universal-translator"
```
Comment (daniejoh) [inline tests/unit/data/test_data.go:6]: Same as with strings earlier. I believe this should not be aliased.

**Review by daniejoh — CHANGES_REQUESTED** [review]

File: keycloak/keycloak.go (hunk 1 above)
Comment (kstigen) [reply keycloak/keycloak.go]: Already done, but forgotten to push :)

File: deployment/kubernetes/client-registration-api.yaml (hunk 5 above)
Comment (kstigen) [reply deployment/kubernetes/client-registration-api.yaml:18]: Yes, but not really. Decided to do this in a separate PR. Have to find out how to do prometheus metrics with gin.

File: keycloak/keycloak.go (hunk 6 above)
Comment (kstigen) [reply keycloak/keycloak.go]: I'm fine with avoiding import aliasing, where we can.

File: tests/unit/data/test_data.go (hunk 7 above)
Comment (kstigen) [reply tests/unit/data/test_data.go:6]: This needs to be aliased, as the package name is `ut` which does not match the last element of the import path.

File: deployment/kubernetes/client-registration-api.yaml (hunk 5 above)
Comment (daniejoh) [reply deployment/kubernetes/client-registration-api.yaml:18]: Okay, I have done the same for now on the agriculture api.

**Review by daniejoh — APPROVED** [review]

**Pull request merged**